File: result/translate/chatgpt/java2python/python/MINIMUM_NUMBER_SUBSETS_DISTINCT_ELEMENTS.py
def f_gold(ar, n):
    res = 0
    ar.sort()
    i = 0
    while i < n:
        count = 1
        while i < n - 1:
            if ar[i] == ar[i + 1]:
                count += 1
            else:
                break
            i += 1
        res = max(res, count)
        i += 1
    return res

File: result/translate/chatgpt/java2cpp/c++/MINIMUM_NUMBER_SUBSETS_DISTINCT_ELEMENTS.cpp
#include <algorithm>

int f_gold(int ar[], int n) {
    int res = 0;
    std::sort(ar, ar + n);
    for (int i = 0; i < n; i++) {
        int count = 1;
        for (; i < n-1; i++) {
            if (ar[i] == ar[i + 1]) count++;
            else break;
        }
        res = std::max(res, count);
    }
    return res;
}